kubernetes-client/src/test/java/io/fabric8/kubernetes/client/dsl/internal/uploadable/PodUploadTest.java

```diff
@@ -89,34 +100,17 @@ void testUploadInvalidParametersShouldThrowException() throws Exception {
   }
 
   @Test
-  void testUploadFileHappyScenarioShouldUploadFile() throws Exception {
-    when(mockContext.getFile()).thenReturn("/mock/dir/file");
+  void testUploadFileHappyScenarioShouldUploadFile() throws IOException, InterruptedException {
     when(mockPathToUpload.toFile())
       .thenReturn(new File(PodUpload.class.getResource("/upload/upload-sample.txt").getFile()));
-    WebSocket.Builder builder = Mockito.mock(WebSocket.Builder.class, Mockito.RETURNS_SELF);
-    when(builder.buildAsync(any())).thenAnswer(newWebSocket -> {
-      final PodUploadWebSocketListener wsl = newWebSocket.getArgument(0, PodUploadWebSocketListener.class);
-      // Set ready status
-      wsl.onOpen(mockWebSocket);
-      wsl.onMessage(mockWebSocket, ByteBuffer.wrap(new byte[] {(byte) 0}));
-      // Set complete status
-      Mockito.doAnswer(close -> {
-        wsl.onClose(mockWebSocket, close.getArgument(0), close.getArgument(1));
-        return null;
-      }).when(mockWebSocket).sendClose(anyInt(), anyString());
-      return CompletableFuture.completedFuture(mockWebSocket);  
-    });
-    when(mockClient.newWebSocketBuilder()).thenReturn(builder);
-
-    final boolean result = PodUpload.upload(mockClient, mockContext, operationSupport, mockPathToUpload);
-
-    assertThat(result, equalTo(true));
+    uploadFileAndVerify(() -> PodUpload.upload(mockClient, mockContext, operationSupport, mockPathToUpload));
     verify(mockPathToUpload, atLeast(1)).toFile();
-    verify(builder, times(1)).uri(argThat(request -> {
-      assertThat(request.toString(), equalTo("https://openshift.com:8443/api/v1/namespaces/default/pods/mock-pod/exec?command=sh&command=-c&command=mkdir+-p+%27%2Fmock%2Fdir%27+%26%26+base64+-d+-+%3E+%27%2Fmock%2Fdir%2Ffile%27&stdin=true&stderr=true"));
-      return true;
-    }));
-    verify(mockWebSocket, atLeast(1)).send(any(ByteBuffer.class));
+  }
+
+  @Test
+  void testUploadFileInputStreamScenarioShouldUploadFile() throws IOException, InterruptedException {
```

Comment:
Is this the right name for this method ?

Reply:
Umm, not sure . What would you suggest? I am okay with renaming this as per your suggestion.

Reply:
It's just a test, naming it's not critical.

Reply:
Maybe `test_upload_file_data_with_ByteArrayInputStream()` WDYT ?

> It's just a test, naming is not critical.
It has nothing to do with test or not. Naming is never critical but it is important as it keeps the code readable and human friendly.

Reply:
Please, just use a `@Display` to create a human readable test description.
e.g. unit of work, state under test, expected behavior, or whatever other convention you like. But make it readable :pray:

Reply:
Some literature: 
- https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html
- https://enterprisecraftsmanship.com/posts/you-naming-tests-wrong/